Automattic/newspack-plugin
# Feat/current campaign group

### All Submissions:






### Changes proposed in this Pull Request:

Together with https://github.com/Automattic/newspack-popups/pull/364 adds the concept of the active campaign group. Only campaigns in the active group will be shown to end users. If there is no active group, no campaigns will be shown. 

**Note that this is a "breaking" change since it means legacy campaigns will stop working as soon as these changes take effect. I'm working on  few possible migration strategies to convert legacy campaign configurations to the new way, including this [CLI-based approach](https://github.com/Automattic/newspack-popups/tree/feat/cli-converter). We'll need to be careful not to deploy these branches until a migration solution is settled on.**

<img width="1420" alt="Screen Shot 2021-01-03 at 3 58 22 PM" src="https://user-images.githubusercontent.com/1477002/103488667-930a4600-4ddc-11eb-8b28-6726fc564588.png">

Dependent PRs:
~https://github.com/Automattic/newspack-plugin/pull/771~
~https://github.com/Automattic/newspack-plugin/pull/769~
~https://github.com/Automattic/newspack-plugin/pull/768~
~https://github.com/Automattic/newspack-plugin/pull/767~

### How to test the changes in this Pull Request:

Switch `newspack-popups` to `feat/current-campaign-group`.

1. Make sure some Campaign Groups have been created. For now, this can only be done in the sidebar of the edit screen for a campaign. 
2. Navigate to Newspack->Campaigns->Campaigns. In the Campaign Groups `SelectControl` select a group. Observe an "Activate" button appears next to it.
3. Click to activate. Observe the label in the `SelectControl` now has "(Active)" after it. Observe there is a "Deactivate" button next to it.
4. Refresh the page, observe the Campaign Groups UI defaults to the active group. Observe this is also true when switching from another tab back to Campaigns.
5. Switch back to "All Campaigns." Assign various Campaign Groups to different campaigns. Note: the work adding Campaign Group UI to the popover is in https://github.com/Automattic/newspack-plugin/pull/767, which is merged into this branch.
6. Experiment with activating various Campaign Groups (including none) and verify in an incognito window that end users only see campaigns in the active group.
7. Verify that any individual campaign can be previewed, whether it is in the active group or not. Verify from both the Campaigns Wizard (click three dots on right->preview) and from within the editor sidebar.
8. Verify that other groups can be previewed successfully, whether active or not. Try this from both the Campaigns and Preview tabs.

### Other information:
